feat:  add C implementation for `stats/base/dists/planck/mean`

yuvimittal opened the pull request:
Resolves #4901

## Description

> What is the purpose of this pull request?

This pull request:

-  Add C implementation for @stdlib/stats/base/dists/planck/mean

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #4901 
## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

yuvimittal committed:
Add C implementation for @stdlib/stats/base/dists/planck/mean

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
Refactor random number generation in JS benchmarks

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mean/README.md:
```diff
@@ -120,6 +120,98 @@ for ( i = 0; i < 10; i++ ) {
+    for ( i = 0; i < 25; i++ ) {
+        lambda = random_uniform( 0.1, 10.0 );
+        y = stdlib_base_dists_planck_mean( lambda );
+        printf( "lambda: %lf, E(X;λ): %lf\n", lambda, y );
```
```suggestion
        printf( "λ: %lf, E(X;λ): %lf\n", lambda, y );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mean/benchmark/benchmark.js:
```diff
@@ -31,14 +32,19 @@ var mean = require( './../lib' );
 	var i;
 
-	lambda = uniform( 100, 0.1, 10.0 );
+	len =100;
+	lambda = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		lambda[ i ] = uniform( 1, 10.0 );
+	}
```
```suggestion
	len = 100;
	lambda = new Float64Array( len );
	for ( i = 0; i < len; i++ ) {
		lambda[ i ] = uniform( 1.0, 10.0 );
	}
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mean/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,67 @@
+	var y;
+	var i;
+
+	len =100;
```
```suggestion
	len = 100;
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mean/src/main.c:
```diff
@@ -0,0 +1,37 @@
+/**
+* Returns the expected value of a Planck distribution with shape parameter `lambda`.
+*
+* @param lambda   shape parameter (must be positive)
+* @return         expected value or NaN if invalid input
```
```suggestion
* @param lambda   shape parameter (must be positive)
* @return         expected value
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mean/examples/c/example.c:
```diff
@@ -0,0 +1,38 @@
+	for ( i = 0; i < 25; i++ ) {
+		lambda = random_uniform( 0.1, 10.0 );
+		y = stdlib_base_dists_planck_mean( lambda );
+		printf( "lambda: %lf, E(X;λ): %lf\n", lambda, y );
```
```suggestion
		printf( "λ: %lf, E(X;λ): %lf\n", lambda, y );
```

Planeshifter commented:
/stdlib lint-autofix

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mean/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,140 @@
+* @param passing   total number of passing tests
+*/
+static void print_summary( int total, int passing ) {
+    printf( "#\n" );
```
File should be TAB-indented.

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mean/src/main.c:
```diff
@@ -0,0 +1,37 @@
+* // returns ~9.5083
+*/
+double stdlib_base_dists_planck_mean( const double lambda ) {
+    if ( stdlib_base_is_nan( lambda ) || lambda <= 0.0 ) {
```
File should be TAB-indented.

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mean/include/stdlib/stats/base/dists/planck/mean.h:
```diff
@@ -0,0 +1,41 @@
+extern "C" {
+#endif
+
+/**
+*Returns the expected value of a Planck distribution with shape parameter `lambda`.
+*
+* @param T - The temperature parameter of the Planck distribution (must be > 0)
+* @return The mean of the Planck distribution for the given temperature
+*/
```
```suggestion
/**
* Returns the expected value of a Planck distribution with shape parameter `lambda`.
*/
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mean/lib/native.js:
```diff
@@ -0,0 +1,62 @@
+* Returns the expected value of a Planck distribution with shape parameter `lambda`.
+*
+* @private
+* @param {number} lambda - shape parameter (must be positive)
```
```suggestion
* @param {PositiveNumber} lambda - shape parameter
```

Planeshifter reviewed (changes requested):
Left a bunch of comments that should be addressed before this PR can move forward.

yuvimittal committed:
feat: change years

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
feat: change years

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal commented:
@Planeshifter , i have made the necessary changes , thank you for reviewing !

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mean/src/main.c:
```diff
@@ -0,0 +1,37 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/expm1.h"
```
```suggestion

#include "stdlib/stats/base/dists/planck/mean.h"
#include "stdlib/math/base/assert/is_nan.h"
#include "stdlib/math/base/special/expm1.h"
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mean/examples/index.js:
```diff
@@ -1,7 +1,7 @@
 * @license Apache-2.0
 *
-* Copyright (c) 2024 The Stdlib Authors.
+* Copyright (c) 2025 The Stdlib Authors.
```
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

Planeshifter committed:
chore: apply suggestions from code review

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
bench: revert changes

Planeshifter committed:
bench: update copyright year and remove unused variable

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mean/benchmark/benchmark.js:
```diff
@@ -31,14 +31,15 @@ var mean = require( './../lib' );
 
 bench( pkg, function benchmark( b ) {
 	var lambda;
+	var len;
```
```suggestion
```

Planeshifter committed:
chore: remove variable declaration

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mean/benchmark/benchmark.js:
```diff
@@ -38,7 +38,7 @@ bench( pkg, function benchmark( b ) {
 	b.tic();
 	for ( i = 0; i < b.iterations; i++ ) {
-		y = mean( lambda[ i % lambda.length ] );
+		y = mean( lambda[ i % 100 ] );
```
```suggestion
		y = mean( lambda[ i % lambda.length] );
```

Planeshifter committed:
chore: avoid number

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mean/benchmark/benchmark.js:
```diff
@@ -38,7 +38,7 @@ bench( pkg, function benchmark( b ) {
 	b.tic();
 	for ( i = 0; i < b.iterations; i++ ) {
-		y = mean( lambda[ i % lambda.length ] );
+		y = mean( lambda[ i % lambda.length] );
```
```suggestion
		y = mean( lambda[ i % lambda.length ] );
```

Planeshifter committed:
chore: add space

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (approved):
Thanks again @yuvi-mittal for your PR and addressing the left comments. Will merge shortly once CI checks have passed again.
